src/web/lib/hakuneko/frontend@classic-light/pages.html

```diff
@@ -327,6 +327,7 @@
              */
             hideViewer( event ) {
                 this.set( 'selectedMedia', -1 );
+                Engine.ChaptermarkManager.addChaptermark( this.selectedChapter );
```

Comment:
It is up to you to decide which actions should update the chapter marker, the following candiates may be possible:

- `showViewer`
- `hideViewer`
- `requestChapterUp`
- `requestChapterDown`

I think your current solution is fine, chapter mark will be updated when the reader is closed (which probably always is the case, except the program is closed while reading or crashes for some reason)